### PR TITLE
[IT-1101] Setup GuarDuty

### PR DESCRIPTION
**Pull request by zaro0508**
Enable guard duty on all accounts and have findings sent to our
securitycentral account.  This is the master member guard duty
configuration[1]

[1] https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_organizations.html

depends on https://github.com/Sage-Bionetworks/aws-infra/pull/288

### CODE REVIEW

File: org-formation/070-guard-duty/_tasks.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  StackDescription: GuardDuty - Trusted IPs bucket
+  DefaultOrganizationBindingRegion: !Ref primaryRegion
+  DefaultOrganizationBinding:
+    Account: !Ref accountId
+  TerminationProtection: true
+  Parameters:
+    resourcePrefix: !Ref resourcePrefix
+    bucketName: !Sub '${resourcePrefix}-${appName}-trusted-ips'
+
+GuardDutyTrustedIps:
```
Comment (ahayden) [inline org-formation/070-guard-duty/_tasks.yaml:51]: I think we should leave this out as its unnecessary

File: org-formation/070-guard-duty/trusted-ips.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+127.0.0.1
```
Comment (ahayden) [inline org-formation/070-guard-duty/trusted-ips.txt:1]: And the `trusted-ips.txt` file can be removed

File: org-formation/070-guard-duty/_tasks.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  RemotePath: !Sub 's3://${resourcePrefix}-${appName}-trusted-ips/trusted_ips.txt'
+  OrganizationBinding:
+    Account: !Ref accountId
+    Region: !Ref primaryRegion
+
+GuardDutyExtension:
+  Type: update-stacks
+  DependsOn:
+    - GuardDuty
+    - GuardDutyTrustedIps
```
Comment (ahayden) [inline org-formation/070-guard-duty/_tasks.yaml:64]: This would get excised too

File: org-formation/070-guard-duty/_tasks.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  StackName: !Sub '${resourcePrefix}-${appName}-extension'
+  StackDescription: GuardDuty - Extension (including IP Set)
+  DefaultOrganizationBindingRegion: !Ref primaryRegion
+  DefaultOrganizationBinding:
+    Account: !Ref accountId
+  TerminationProtection: true
+  Parameters:
+    resourcePrefix: !Ref resourcePrefix
+    detectorId: !CopyValue [ !Sub '${resourcePrefix}-${appName}-detector-id' ]
+    bucketName: !CopyValue [ !Sub '${resourcePrefix}-${appName}-trusted-ips-bucket-name' ]
```
Comment (ahayden) [inline org-formation/070-guard-duty/_tasks.yaml:75]: Not needed

File: org-formation/070-guard-duty/_tasks.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    Region: !Ref primaryRegion
+
+GuardDutyExtension:
+  Type: update-stacks
+  DependsOn:
+    - GuardDuty
+    - GuardDutyTrustedIps
+  Template: https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra/master/templates/GuardDuty/guard-duty-extension.yaml
+  StackName: !Sub '${resourcePrefix}-${appName}-extension'
+  StackDescription: GuardDuty - Extension (including IP Set)
```
Comment (ahayden) [inline org-formation/070-guard-duty/_tasks.yaml:67]: Then this would be `GuardDuty Extension`

File: org-formation/070-guard-duty/_tasks.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      ExcludeAccount:
+        - !Ref accountId
+    AllBinding:
+      Account: '*'
+      IncludeMasterAccount: true
+  Parameters:
+    resourcePrefix: !Ref resourcePrefix
+    accountId: !Ref accountId
+
+# bucket that contains the trusted IP addresses that are excluded from GuardDuty findings
```
Comment (ahayden) [inline org-formation/070-guard-duty/_tasks.yaml:37]: I suggest not adding any trusted ips list

**Review by ahayden — APPROVED** [review]
I think it's best to not include a trusted ips list, but it's up to you if you want to try it. `127.0.0.1` might be harmless today, though that might not always be the case as AWS is adding new analysis that might include loopback on EC2 or container runners, etc. at some point.

File: org-formation/070-guard-duty/_tasks.yaml (hunk 6 above)
Comment (zaro0508) [reply org-formation/070-guard-duty/_tasks.yaml:37]: i think it's nice to have it as an option.  I mean if you don't put anything on the list then it's basically the same as not having it there at all.

File: org-formation/070-guard-duty/_tasks.yaml (hunk 1 above)
Comment (zaro0508) [reply org-formation/070-guard-duty/_tasks.yaml:51]: it's optional.  it doesn't do anything if you don't put any values into trusted-ips.txt file.

**Pull request merged**